autotest/gdrivers/vrtprocesseddataset.py

```diff
@@ -1198,6 +1198,123 @@ def test_vrtprocesseddataset_trimming_errors(tmp_vsimem):
         )
 
 
+###############################################################################
+# Test expressions
+
+
+@pytest.mark.parametrize(
+    "expression,src,expected,error",
+    [
+        pytest.param(
+            "return [ALL_BANDS[1], ALL_BANDS[2]]",
+            np.array([[[1, 2]], [[3, 4]], [[5, 6]]]),
+            np.array([[[3, 4]], [[5, 6]]]),
+            None,
+            id="multiple bands in, multiple bands out (1)",
+        ),
+        pytest.param(
+            "return [ALL_BANDS]",
+            np.array([[[1, 2]], [[3, 4]], [[5, 6]]]),
+            np.array([[[1, 2]], [[3, 4]], [[5, 6]]]),
+            None,
+            id="multiple bands in, multiple bands out (2)",
+        ),
+        pytest.param(
+            """
+             var chunk[10];
```

Comment:
That's interesting... from a security point of view.
- can that trigger arbitrary memory allocation ?
- does exprtk does bound checking when accessing the arrays ?
I suspect you could also cause infinite /very long looping ?

Reply:
> can that trigger arbitrary memory allocation

Array sizes can be up to 2 billion, so you can cause a large allocation with:

```             
var out[2000000000];
return [out];
```

> does exprtk do bound checking when accessing the arrays ?

Not by default, but enabled in fd94c9e0ed1a10edf90031312128f5826d8e1cdf

Reply:
> Array sizes can be up to 2 billion, so you can cause a large allocation with:

could we improve exprtk to offer a way to limit the allocations?

Reply:
> could we improve exprtk to offer a way to limit the allocations?

@rouault: Yes the maximum vector size that can be created in the expression can be set explicitly:

https://www.partow.net/programming/exprtk/readme.html#1473


```
parser_t parser; 
parser.settings().set_max_local_vector_size(12345); 

```